src/managers/WebhookManager.js

```diff
@@ -37,23 +38,111 @@ class WebhookManager {
   _onNewWebhookEvent = (webhook: Webhook): (event: Event) => void =>
     (event: Event) => {
       try {
-        const responseHandler = (error, response) => {
+        const defaultWebhookVariables = {
+          // todo add old defaults for compatibility
+          PARTICLE_DEVICE_ID: event.deviceID,
+          PARTICLE_EVENT_NAME: event.name,
+          PARTICLE_EVENT_VALUE: event.data,
+          PARTICLE_PUBLISHED_AT: event.publishedAt,
+        };
+
+        let eventDataVariables = {};
+        if (typeof event.data === 'string') {
+          try {
+            eventDataVariables = JSON.parse(event.data);
+          } catch (error) {
+            eventDataVariables = {};
+          }
+        }
+
+        const webhookVariablesObject = webhook.noDefaults
+          ? eventDataVariables
+          : {
+            ...defaultWebhookVariables,
+            ...eventDataVariables,
+          };
+
+        const requestJSON = webhook.json && JSON.parse(
+            hogan
+              .compile(JSON.stringify(webhook.json))
+              .render(webhookVariablesObject),
+          );
+
+
+        const requestFormData = webhook.form && JSON.parse(
+            hogan
+              .compile(JSON.stringify(webhook.form))
+              .render(webhookVariablesObject),
+          );
+
+        const requestUrl = hogan
+          .compile(webhook.url)
+          .render(webhookVariablesObject);
+
+        const requestQuery = webhook.query && JSON.parse(
+            hogan
+              .compile(JSON.stringify(webhook.query))
+              .render(webhookVariablesObject),
+          );
+
+        const responseTopic = webhook.responseTopic && hogan
+          .compile(webhook.responseTopic)
+          .render(webhookVariablesObject);
+
+        const errorResponseTopic = webhook.errorResponseTopic && hogan
+          .compile(webhook.responseTopic)
+          .render(webhookVariablesObject);
+
+        const responseHandler = (
+          error: ?Error,
+          response: http$IncomingMessage,
+          responseBody: string | Buffer | Object,
+        ) => {
           if (error) {
-            // todo block the webhook calls after some amount of fails
+            // todo block the webhook calls after 10 fails
             // on 1 min or so..
-            // todo responseTemplates
+            if (errorResponseTopic) {
```

Comment:
I didn't know that this existed. I though `responseTopic` was emitted whether the hook requests succeeds or not.